CMakeLists.txt

```diff
@@ -315,9 +315,18 @@ add_custom_target(ALL_MULTILANG)
 add_dependencies(ALL_FIRMWARE ALL_ENGLISH ALL_MULTILANG)
 set_target_properties(ALL_MULTILANG PROPERTIES EXCLUDE_FROM_ALL FALSE)
 
+# Ensure the source tree doesn't contain another Configuration_prusa.h
+include(cmake/Check_wrong_config.cmake)
+
+add_custom_target(
+  test_wrong_config COMMAND ${CMAKE_COMMAND} -DPROJECT_SOURCE_DIR=${PROJECT_SOURCE_DIR} -P
+                            ${PROJECT_CMAKE_DIR}/Check_wrong_config.cmake
+  )
+
 function(add_base_binary variant_name)
   add_executable(${variant_name} ${FW_SOURCES} ${FW_HEADERS} ${VARIANT_CFG_DST})
   set_target_properties(${variant_name} PROPERTIES EXCLUDE_FROM_ALL TRUE)
+  # add_dependencies(${variant_name} test_wrong_config) # Uncomment to check at build time
```

Comment:
We should probably enable this - otherwise someone could configure with cmake, (passing the configure-time check) and then run pf build and break the build without knowing why.

Reply:
There's probably a way to prevent this by specifying a dummy output/byproducts file for cmake to look at so that it won't re-run unless other dependencies have changed. Maybe there's even a way to key it off the presence of the configuration_prusa.h file :thinking:

Reply:
Thought: a little clunky but perhaps it could be e.g. a post-build command for the elf - sure, it'll still build the incorrect binary, and throw an error after that. But this way if the elf hasn't changed, it won't rerun the check either leaving it a true no-op.

Reply:
It also doesn't work the way I would like, as it runs late. This target would need to be a dependency on every object.

Reply:
Using a target with a dummy would be better. If you attach a post-build command it's going to run for all the variants. At least with a single global target you get just one run.